Common/src/main/java/com/agilepro/commons/UserRole.java

```diff
@@ -272,6 +293,66 @@
 	@Label("CampaignType DeleteAll") CAMPAIGNTYPE_DELETE_ALL(false, CAMPAIGN_VIEW),
 
 	/**
+	 * The project view.
+	 **/
+	@Label("Projects view") PROJECT_VIEW,
+
+	/**
+	 * The project edit.
+	 **/
+	@Label("Projects Edit") PROJECT_EDIT(false, PROJECT_VIEW, PROJECT_GROUP),
+
+	/**
+	 * The project delete.
+	 **/
+	@Label("Projects Delete") PROJECT_DELETE(false, PROJECT_VIEW),
+
+	/**
+	 * The project delete all.
+	 **/
+	@Label("Projects DeleteAll") PROJECT_DELETE_ALL(false, PROJECT_VIEW),
+
+	/**
+	 * The project members view.
+	 **/
+	@Label("Project Members view") PROJECT_MEMBER_VIEW,
+
+	/**
+	 * The project members edit.
+	 **/
+	@Label("Project Members Edit") PROJECT_MEMBER_EDIT(false, PROJECT_MEMBER_VIEW, MEMBER_GROUP),
+
+	/**
+	 * The project members delete.
+	 **/
+	@Label("Project Members Delete") PROJECT_MEMBER_DELETE(false, PROJECT_MEMBER_VIEW),
+
+	/**
+	 * The project members delete all.
+	 **/
+	@Label("Project Members DeleteAll") PROJECT_MEMBER_DELETE_ALL(false, PROJECT_MEMBER_VIEW),
+
+	/**
+	 * The tags view.
+	 **/
+	@Label("Tags View") TAG_VIEW,
+
+	/**
+	 * The tags edit.
+	 **/
+	@Label("Tags Edit") TAG_EDIT(false, TAG_VIEW, PROJECT_GROUP),
+
+	/**
+	 * The tags delete.
+	 **/
+	@Label("Tags Delete") TAG_DELETE(false, TAG_VIEW),
+
+	/**
+	 * The tags delete all.
+	 **/
+	@Label("Tags DeleteAll") TAG_DELETE_ALL(false, TAG_VIEW),

```

Comment:
Dont create DELETE_ALL roles, that is meant only for testing. So use TEST role for deleteAll() 

And remove ur DELETE_ALL roles